Refresh app to SDK 3.1.7

osfameron opened the pull request:
These changes make the app compile and run with `dotnet run` in the project directory... (e.g. haven't actually verified that it works to spec)

 * updated to
    - SDK 3.1.7
    - netcoreapp3.1
    - C# 9
    - add global.json to ensure right netcoreapp SDK used
 * remove Couchbase.Services (e.g. just Couchbase.Query etc)
 * add Couchbase.KeyValue etc. as appropriate
 * `parameters` object has been removed, add to Options instead
 * UseStreaming removed (not sure if this has been renamed?)
 * IQueryResult has changed
 * ICouchbaseCollection renamed
 * Cluster.ConnectAsync call changed
 * tidy up context string templating
 * await foreach
 * LookupInSpec list in LookupInAsync
 * followed ASP.NET Core 2.2 -> 3.0 migration as per instructions in
   https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30

**Draft** pull request for @AV25242 and others to point out any major issues with this approach so far.

Next steps will be to verify app actually *works*, update to the new spec, add docker-compose etc. to pair with the updated frontend.

osfameron committed:
Refresh app to SDK 3.1.7

These changes make the app compile and run with `dotnet run` in the project directory...

 * updated to
    - SDK 3.1.7
    - netcoreapp3.1
    - C# 9
    - add global.json to ensure right netcoreapp SDK used
 * remove Couchbase.Services (e.g. just Couchbase.Query etc)
 * add Couchbase.KeyValue etc. as appropriate
 * `parameters` object has been removed, add to Options instead
 * UseStreaming removed (not sure if this has been renamed?)
 * IQueryResult has changed
 * ICouchbaseCollection renamed
 * Cluster.ConnectAsync call changed
 * tidy up context string templating
 * await foreach
 * LookupInSpec list in LookupInAsync
 * followed ASP.NET Core 2.2 -> 3.0 migration as per instructions in
   https://docs.microsoft.com/en-us/aspnet/core/migration/22-to-30

maria-robobug commented on try-cb-dotnet/Services/CouchbaseService.cs:
```diff
@@ -1,30 +1,33 @@
-                        .WithServers("couchbase://10.112.193.101")
-                        .WithBucket("travel-sample")
-                        .WithCredentials("Administrator", "password")
+                 var options = new ClusterOptions
```
I think you have an extra space before `var`

maria-robobug commented on try-cb-dotnet/Services/HotelService.cs:
```diff
@@ -50,9 +51,9 @@ public async Task<IEnumerable<dynamic>> FindHotel(string description, string loc
                 ));
             }
-            var Q = new SearchQuery { Query = query };
+            // var Q = new SearchQuery { Query = query };
```
How come this is commented out?

maria-robobug commented on try-cb-dotnet/Services/HotelService.cs:
```diff
@@ -50,9 +51,9 @@ public async Task<IEnumerable<dynamic>> FindHotel(string description, string loc
+            // var Q = new SearchQuery { Query = query };
 
-            var queryString = (Q).ToJson();
+            // var queryString = (Q).ToJson();
```
Same here

osfameron replied:
I think the API's changed, I commented as wasn't quite sure if this would need to be reintroduced, but I'll tidy up when I get to the next step (e.g. actually running the code and checking it works :-) )

osfameron replied:
For the record, this code *doesn't work*, but it looks like it didn't work previously (as per commit history), so this will be fun to debug :D

maria-robobug reviewed (approved):
Limited in my C# knowledge but LGTM, just some minor comments! I assume there will be a different PR for agent scope updates etc?

osfameron committed:
Dockerfile

Note the ASPNETCORE_URLS workaround, to expose :8080 on
localhost. This still gives the warning:

  warn: Microsoft.AspNetCore.Server.Kestrel[0]
      Unable to bind to http://localhost:8080 on the IPv6 loopback
      interface: 'Cannot assign requested address'.

But the binding actually works. Setting 0.0.0.0 or 127.0.0.1 would
work without a warning message, but the experience seems more confusing
with regards to log messages, or running the app locally with `dotnet run`

osfameron committed:
Fix controllers, remove static frontend

osfameron committed:
Add Swagger-ui support

osfameron committed:
Fix resolution of Couchbase Cluster service

 * Point Cluster to localhost for now. (This will need to be configurable)
 * Fix error handling so that an error is thrown on failure to connect.

osfameron committed:
Date-formatting flightPaths

Minor update to flightPaths route:

 * parse US date mm/dd/yyyy as per spec
 * make controller path explicit
	(the `[controller]` magic feels unnecessary and confusing, relative
	to our other sample apps)

osfameron committed:
Make route explicit

osfameron committed:
Update Airports query

 * Scope to inventory
 * tidy to use Rows.ToListAsync()
 * show the explicit parameter list in context
 (e.g. the query itself was already safe, but avoid even using
  string replace in the output to demonstrate best practice)

osfameron committed:
FlightService - scoped query

 * use scoped query
 * update context
 * tidy conversion to list

osfameron committed:
Update Hotel search (WIP)

Not working, but I think as per
4eeadcf8c101fa62bfeda08a83a03bfdbfd3fa49
that this has never worked. Will come back to it.

osfameron committed:
User signup.

 * Moved under tenants/{tenant}
 * change to System.Text.Json as the NewtonSoft JsonProperty attribute doesn't work
 * scoped KV call

osfameron committed:
Add CORS for frontend service

osfameron committed:
User service tenant aware

osfameron committed:
Move all models to System.Text.Json
a2fa27c Add CORS for frontend service

osfameron committed:
Flight booking

osfameron committed:
Fix hotel search

osfameron committed:
First stage - dockerisation (WIP)

osfameron committed:
README and config

 * Note that config is done non-idiomatically, will seek guidance from our .NET specialists
 * README copied from Node.js and tweaked - needs update for specifics of config as above

osfameron committed:
Update Docker compose files

osfameron committed:
Update README.md with .NET details

osfameron committed:
Narration updated

Return context to match behaviour of other sample apps.
